### DISCUSSION TITLE
Livewiere update route registered twice when defining custom update route (and other route related issues)

### DISCUSSION THREAD

**Discussion by Sn0wCrack** [livewire/livewire#7283] [7. Bugs (archived)]
### Livewire version

v3.1.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

**Overview:**

It seems like when using `Livewire::setUpdateRoute` as [described in the Livewire documentation](https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint) that two Livewire update routes are registered, one initially from within the `HandleRequests::handleUpdate` function from within Livewire, and the second time from within the route file that `Livewire::setUpdateRoute` is called in.
e.

When there is no route cache file this presents no real issues, the Livewire update route specified in the application route is used by Livewire, however after running `php artisan route:cache` the original `/livewrire/update` route is now used by Livewire to run its API calls, rather than the route specified in the application's route file.

**Reproduction:**

With a basic Livewire installation running `php artisan route:list | grep livewire` will return the following:

```
  GET|HEAD        livewire/livewire.js ................................................................................................................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD        livewire/preview-file/{filename} ....................................................................................... livewire.preview-file › Livewire\Features › FilePreviewController@handle
  POST            livewire/update ............................................................................................................. livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  POST            livewire/upload-file ..................................................................................................... livewire.upload-file › Livewire\Features › FileUploadController@handle
```

Add the following to `routes/web.php`:

```php
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/_internal/livewire/update', $handle)
         ->name('site.livewire.update'); // Set a name to avoid collision with existing route
});
```

The output of `php artisan route:list | grep livewire` is now:

```
  POST            _internal/livewire/update .............................................................................................. site.livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  GET|HEAD        livewire/livewire.js ................................................................................................................ Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD        livewire/preview-file/{filename} ....................................................................................... livewire.preview-file › Livewire\Features › FilePreviewController@handle
  POST            livewire/update ............................................................................................................. livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  POST            livewire/upload-file ..................................................................................................... livewire.upload-file › Livewire\Features › FileUploadController@handl
```

In this state Livewire components will hit `/_internal/livewire/update` when making requests.

Run `php artisan route:cache`

The output of `php artisan route:list | grep livewire` is now:

```
  POST            _internal/livewire/update .............................................................................................. site.livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  GET|HEAD        livewire/livewire.js .................................................................................. generated::uiNJMO6KDaXVnrZj › Livewire\Mechanisms › FrontendAssets@returnJavaScriptAsFile
  GET|HEAD        livewire/preview-file/{filename} ....................................................................................... livewire.preview-file › Livewire\Features › FilePreviewController@handle
  POST            livewire/update ............................................................................................................. livewire.update › Livewire\Mechanisms › HandleRequests@handleUpdate
  POST            livewire/upload-file ..................................................................................................... livewire.upload-file › Livewire\Features › FileUploadController@handle
```

In this state Livewire components will hit `/livewire/update` when making requests.

**Comments:**

Beyond the issue caused when running `php artisan route:cache` this will still lead to an additional endpoint being created that is still usable and potentially may cause collisions with user specified routes.

I've also noticed it's not possible to customise following routes at all (or at least it's not documented):

- livewire/livewire.js
- livewire/preview-file/{filename}
- livewire/upload-file

It would be good to able to customise or disable the above routes when they're not going to be used in an application.

This also seems to be related to: https://github.com/livewire/livewire/discussions/6888

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by galliroberto** [comment]
Do you have solved?

**Reply by Sn0wCrack** [reply]
This is still an issue unfortunately and is a bug with Livewire itself and how setting the route works.
